satpy/readers/satpy_cf_nc.py

```diff
@@ -329,8 +334,18 @@ def get_area_def(self, dataset_id):
             raise NotImplementedError
 
 
+def _datetime_parser(json_dict):
+    import dateutil.parser
```

Comment:
This is a new dependency isn't it?

Reply:
Yes, see PR description. I'm open to discuss this (parsing timestamps with varying precision, e.g. "2020-01-01 00:00" or "2020-01-01 00:00.00000")

Reply:
Found a way using `datetime.fromisoformat`

Reply:
Ah sorry I missed that note in the description. Looks like the supported formats was improved in Python 3.11: https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat

Do you think that breaks anything or changes how you feel about your solution? I'm not sure I understand the note about YYYY-MM-DD. Is it saying YYYYMMDD didn't used to work but now it does?

Reply:
Oh, good that you double checked it. I missed that hint in the documentation. This is what works with both Python-3.10 and 3.11:

```
In [2]: dt.datetime.fromisoformat("2000-01-01")
Out[2]: datetime.datetime(2000, 1, 1, 0, 0)

In [3]: dt.datetime.fromisoformat("2000-01-01 12:00")
Out[3]: datetime.datetime(2000, 1, 1, 12, 0)

In [4]: dt.datetime.fromisoformat("2000-01-01 12:15:23")
Out[4]: datetime.datetime(2000, 1, 1, 12, 15, 23)

In [5]: dt.datetime.fromisoformat("2000-01-01 12:15:23.123456")
Out[5]: datetime.datetime(2000, 1, 1, 12, 15, 23, 123456)
```

The CF writer encodes datetime objects with their string representation, so I think this covers all cases.

In Python-3.11 you can also do this:
```
In [4]: dt.datetime.fromisoformat("20000101")
Out[4]: datetime.datetime(2000, 1, 1, 0, 0)

In [5]: dt.datetime.fromisoformat("20000101T1200Z")
Out[5]: datetime.datetime(2000, 1, 1, 12, 0, tzinfo=datetime.timezone.utc)

In [6]: dt.datetime.fromisoformat("2000-01-01T12:00Z")
Out[6]: datetime.datetime(2000, 1, 1, 12, 0, tzinfo=datetime.timezone.utc)
```

Reply:
Alternatively, we could of course update the CF writer to always encode datetime objects with microsecond precision.

Reply:
I guess as long as the writer produces something the reader can parse in both versions of Python I'm OK with it. Users using 3.11+ will just get extra flexibility so they can read other input files.

Reply:
Yes I think that's the case